src/libs/IOUUtils.ts

```diff
@@ -86,18 +87,26 @@ function updateIOUOwnerAndTotal<TReport extends OnyxInputOrEntry<Report>>(
 
     // Let us ensure a valid value before updating the total amount.
     iouReportUpdate.total = iouReportUpdate.total ?? 0;
+    iouReportUpdate.unheldTotal = iouReportUpdate.unheldTotal ?? 0;
 
     if (actorAccountID === iouReport.ownerAccountID) {
         iouReportUpdate.total += isDeleting ? -amount : amount;
+        if (!isOnhold) {
+            iouReportUpdate.unheldTotal += isDeleting ? -amount : amount;
+        }
     } else {
         iouReportUpdate.total += isDeleting ? amount : -amount;
+        if (!isOnhold) {
+            iouReportUpdate.unheldTotal += isDeleting ? amount : -amount;
+        }
```

Comment:
I've updated the optimistic unheldTotal for IOU too. However, there is a problem.

1. User A submit $2 and $2 to user B (total = 4, unheldTotal = 4, owner = user A)
2. User A hold $2 (total = 4, unheldTotal = 2, owner user A)
3. User B submit $10

When user B submits, it will go into the `else` block, subtracting `total` from the amount (10). So, the new total will be -6. If it's < 0, we switch the owner with this logic and switch the sign too, so the new optimistic total is 6.
https://github.com/Expensify/App/blob/866ce4b3957111d2c3ea5a7dccc3c86b90ed4e7c/src/libs/IOUUtils.ts#L96-L100

The problem I see is, that BE still keeps the sign, so the `total` from BE is -6. Also, the `unheldTotal` is -8, which looks like it comes from `unheldTotal` (2) - amount (10). So, I apply this calculation for the optimistic one too as you can see above. I'm not sure if the `unheldTotal` should be -8, but at least I match it with the BE. I think it will be another big problem to solve and decide the expected behavior.

Notice the image below shows Pay -8 (unheldTotal).

<img width="1199" alt="image" src="https://github.com/user-attachments/assets/e0493330-5617-4af2-8bf0-246aaf497eaa">

Reply:
Hmm this looks tricky, i think we need to agree on the expected behaviour on what the total and the unheldTotal should be in this case and also in the case where we unhold the $2 request. (can we still unhold that request? and how can do it?)

Reply:
We can still unhold it and the `unheldTotal` will be the same as `total` (-6). If I hold it again, it will be -8 again.

> User A submit $2 and $2 to user B (total = 4, unheldTotal = 4, owner = user A)
> User A hold $2 (total = 4, unheldTotal = 2, owner user A)
> User B submit $10

I think if we have this case, the new total should be 6 (number is correct already in prod, but it's in negative) and unheldTotal should be:

current held = total (4) - unheldTotal (2) = 2
new unheldTotal = new total (6) - current held (2) = 4

I suspect the BE already did this calculation, but because the total is negative (-6), subtracting with 2 will be -8.

cc: @robertjchen

Reply:
Is the problem around IOUs and expense reports having different signage? 🤔

Reply:
I think the problem is more related to the fact that both users can submit expenses in the same iou report. In the above scenario the unheldTotal will be grater than the total

total: $6
unheldTotal: $8

Reply:
> I think the problem is more related to the fact that both users can submit expenses in the same iou report.

I think this is expected.

> In the above scenario the unheldTotal will be grater than the total

And both are negative.

> unheldTotal on the backend is simply the reimbursable total (excluding held transactions) + non reimbursable total (excluding held transactions).

From the step I performed above, there is no non-reimbursable, so the wrong calculation is from the reimbursable total. 

> Is the problem around IOUs and expense reports having different signage? 🤔

I think the real problem here is only related to the IOU `total` that is negative after user B adds an expense that the amount is bigger than the current `total`, as I mentioned above.

> I suspect the BE already did this calculation, but because the total is negative (-6), subtracting with 2 will be -8.

The `total` is -6 and the held amount is 2. If the `total` is 6, then the issue will be gone, probably.

Reply:
@robertjchen Any updates on this? Do you know what should be the expected values for `total` and `unheldTotal` in this case?

Reply:
> The total is -6 and the held amount is 2. If the total is 6, then the issue will be gone, probably.

IOU and expense report amounts have different signs, and I don't see what else we can change on the backend 🤔 Any discrepancies would have to be accounted for on the FE: could we use absolute values on the totals before performing the calculations? Would that solve the issue here? (We can then properly add the sign back if needed according to the type of the report).

Reply:
> Any discrepancies would have to be accounted for on the FE: could we use absolute values on the totals before performing the calculations?

The `unheldTotal` is also being calculated on the BE, so even if we do this on the FE, the value from the BE will still be wrong. Btw, this is an IOU issue only. Maybe the BE needs to apply the same logic as this?
https://github.com/Expensify/App/blob/2463bf74b4edabdde887636f7874fdbbfcf9eb2c/src/libs/IOUUtils.ts#L97-L102

Reply:
To clarify, do we want the `total` amount to be negative for IOUs when it's returned from the backend? (or are we referring to `unheldTotal`) 🤔

Reply:
We want both `total` and `unheldTotal` to be positive for IOUs. Currently, it's both negative, but based on my previous comments, my guess is that `unheldTotal` is negative because the `total` is negative. If we can fix the `total`, then `unheldTotal` will hopefully be fixed too.

Reply:
I can't change the report `total` on the backend, as many things rely on it and it has been the same for every expense report and IOU for many years.

On the FE, we check the report type and reverse the `total` depending on whether it's an IOU or not, can this logic be applied here as well?

https://github.com/Expensify/App/blob/2463bf74b4edabdde887636f7874fdbbfcf9eb2c/src/libs/ReportUtils.ts#L7699-L7722

Reply:
`getNonHeldAndFullAmount` already have that logic.

> const coefficient = isExpenseReport(iouReport) ? -1 : 1; 

We can use `Math.abs` but that still won't fix the `unheldTotal` wrong value issue.

> In the above scenario the unheldTotal will be grater than the total
>
> total: $6
> unheldTotal: $8

expected is $4.

Reply:
Why $4?
- A request $2
- A request $2 (on hold)
- B request $10

Now A has to pay B either:
1. $8
2. $6

Reply:
Hmm, actually, after thinking about it the 2nd time, I think `unheldTotal` should be $6. I think User A holding his expenses shouldn't affect the `unheldTotal`. The IOU interactions are really tricky 🤯

Reply:
@bernhardoj Can you make the required changes (if any) so that we have:
1. total $6 (or -$6)
2. unheldTotal $8 (or -$8)

The unheldTotal (absolute value) is greater than the total which is weird but given that the owner changes this is still logical. Now, the thing is that after A unhold his request, we will have to subtract that value from unheldTotal which is again also weird as usually we sum that value up.

One more thing, we need to know when to sum/subtract, consider this scenario:

- A request $2
- A request $2 (on hold)
- B request $10
- B request $10 (on hold)

The total should be: $16 (absolute value)
The unheldTotal should be: $8 (absolute value)

If A unhold his request the unheldTotal should be $6 (subtract)
If B unhold his request the unheldTotal should be $18 (sum)

So it seems that depending on the owner we can determine if we we should add/subtract to/from unheldTotal

Reply:
Let me know if that makes sense

Reply:
I'm starting to believe the unheldTotal is bigger is correct, but yeah, that's how BE works, and the optimistic data is already works that way too. (except it's in positive because of the logic in the conversation below)

https://github.com/user-attachments/assets/23e9b10f-b2ec-4eb6-a072-c58b1e7ba214

Reply:
Okay great, so what needs to be done so we can move forward, just the sign right?

Reply:
@Gonals May be able to help if that logic is no longer true?

Reply:
AFAIK, we haven't changed that, but it has been a loooong while since I touched that area

Reply:
@robertjchen Would you be able to check again so we can move forward here?

Reply:
Sorry for the delay- I think switching the sign optimistically as you've proposed may be necessary, as there won't be any further changes to the amounts being returned from the BE 😥